### ISSUE TITLE
"Too many file descriptors" error on Windows

### ISSUE THREAD

**Issue by ichorid** [Tribler/tribler#5435]
7.5.1 on Windows, from reporter:

```python
Traceback (most recent call last):
File "src\tribler-gui\tribler_gui\core_manager.py", line 64, in on_core_finished
RuntimeError: The Tribler core has unexpectedly finished with exit code -1 and status: 0!

Traceback (most recent call last):
File "src\run_tribler.py", line 115, in <module>
File "src\run_tribler.py", line 99, in start_tribler_core
File "lib\asyncio\base_events.py", line 534, in run_forever
File "lib\asyncio\base_events.py", line 1735, in _run_once
File "lib\selectors.py", line 323, in select
File "lib\selectors.py", line 314, in _select
ValueError: too many file descriptors in select()
```

The answer is [instantly available on SO](
https://stackoverflow.com/questions/47675410/python-asyncio-aiohttp-valueerror-too-many-file-descriptors-in-select-on-win): 

>By default Windows can use only 64 sockets in asyncio loop. This is a limitation of underlying select() API call.
To increase the limit please use ProactorEventLoop.

**Comment by egbertbouman** [comment]
Using the `ProactorEventLoop` event loop will break Tribler on Python 3.7 and lower, as it doesn't support `create_datagram_endpoint`. In Python 3.8 that limitation if fixed, and `ProactorEventLoop` is the default on Windows.

**Comment by ichorid** [comment]
Well, this means we have to move our infrastructure to python 3.8

**Comment by devos50** [comment]
Our macOS builds already include Python 3.8. Our 64-bit Windows build is still Python 3.7(.5?) and I'm not sure about our 32-bit build. If we also upgrade our Linux builder, moving to Python 3.8 is more than doable. Do we want this for the 7.5.2 release?

**Comment by ichorid** [comment]
> Do we want this for the 7.5.2 release?

Yes, that's the perfect opportunity to do this. Everything else is stable, so no surprises (except for 3.8-specific :gift: of course).

**Comment by devos50** [comment]
There are always surprises when touching the builder infrastructure 🎁. With Tribler, it's always Christmas!

I'll try to spend some time on the upgrade to Python 3.8.

**Comment by hbiyik** [comment]
shamelessly advertising:

https://github.com/hbiyik/triblerd-buildsys

python 3.8.1 ready :)

**Comment by ichorid** [comment]
@hbiyik , we would like to streamline our build system, but definitely not for a minor release ))

**Comment by ichorid** [comment]
Fixed automatically by using Python 3.8 on Windows

**Comment by ichorid** [comment]
Never seen it since

**Comment by qstokkink** [comment]
I was highly confused as to how Sentry seemed to suggest that we were using select on Windows even though the `ProactorEventLoop` should be the default, then I found this:

https://github.com/Tribler/tribler/blob/afa20e567cadf8d272fad082a109be2f7e263783/src/tribler/core/start_core.py#L166-L169

It's worth revisiting if the reason for setting `SelectorEventLoop` on Windows has been resolved (#5485) for newer Python versions.
